sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OrchestrationController.java

```diff
@@ -71,8 +76,8 @@ OrchestrationResponse template() {
     final var inputParams = Map.ofEntries(templateVariable.apply("german"));
 
     final var prompt =
-        new OrchestrationPrompt(inputParams)
-            .withTemplate(TemplateConfig.fromMessages(templateMessage));
+        new OrchestrationPrompt(
+            inputParams, config.withTemplate(TemplateConfig.fromMessages(templateMessage)));
```

Comment:
I started writing this message after my previous [comment](https://github.com/SAP/ai-sdk-java/pull/127/files#r1827819209).

I feel an OrchestrationPrompt is another config that knows prompt templates in this case.

```
// update config
config.withTemplate(TemplateConfig.fromMessages(templateMessage));
// send a request with an inputParams + a config
return client.chatCompletion(inputParams, config);
```

Reply:
> It's somewhat funny that you are suggesting an API usage that would exactly not work 😄

You got my point 😄  which is similar to Alex's suggestion.